### PR TITLE
server: check the store version when changing tombstone status

### PR DESCRIPTION
**Pull request by HunDunDM**
Signed-off-by: Zheng Xiangsheng <hundundm@gmail.com>

### What problem does this PR solve?

* When changing a `Tombstone` store to `Up` or `Offline`, if the `cluster-version` has been upgraded at this time, the operation should fail.

### What is changed and how it works?

* The store of lower version is no longer allowed to change from `Tombstone` back to `Up`.
* Fix the bug that restarting the store will cause store-limit and statistics to be reset.

### Check List

Tests

- Unit test

Related changes

- Need to cherry-pick to the release branch

### Release note

- The store of lower version is no longer allowed to change from `Tombstone` back to `Up`.
- Fix the bug that restarting the store will cause store-limit and statistics to be reset.

### CODE REVIEW

File: server/statistics/store.go [hunk 1]
```diff
@@ -45,7 +45,9 @@ func NewStoresStats() *StoresStats {
 func (s *StoresStats) CreateRollingStoreStats(storeID uint64) {
 	s.Lock()
 	defer s.Unlock()
-	s.rollingStoresStats[storeID] = newRollingStoreStats()
+	if _, ok := s.rollingStoresStats[storeID]; !ok {
```
Comment (lhy1024) [inline server/statistics/store.go:48]: we should remove `CreateRollingStoreStats`，only to use `GetOrCreateRollingStoreStats`

**Comment by lhy1024** [comment]
It also is a part of #3201

**Review by disksing — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

File: server/statistics/store.go (hunk 1 above)
Comment (HunDunDM) [reply server/statistics/store.go]: lgtm

**Pull request merged**